tests/phases/design/test_electrical_design.py

```diff
@@ -0,0 +1,276 @@
+        config["export_system_design"][k] = v
+
+        o = ElectricalDesign(config)
+        o.run()
+        cost = o.total_cost
+
+        assert cost != base_cost
+        
+config = extract_library_specs("config", "export_design")
+
+def test_export_system_creation():
+    export = ElectricalDesign(config)
+    export.run()
+
+    assert export.num_cables
+    assert export.length
+    assert export.mass
+    assert export.cable
+    assert export.total_length
+    assert export.total_mass
+    assert export.num_substations
+    assert export.topside_mass
+    assert export.substructure_mass
+
+def test_number_cables():
+    export = ElectricalDesign(config)
+    export.run()
+
+    assert export.num_cables == 11
+
+def test_cable_length():
+    export = ElectricalDesign(config)
+    export.run()
+
+    length = (0.02 + 3 + 30) * 1.01
+    assert export.length == length
+
+def test_cable_mass():
+    export = ElectricalDesign(config)
+    export.run()
+
+    length = (0.02 + 3 + 30) * 1.01
+    mass = length * export.cable.linear_density
+    assert export.mass == pytest.approx(mass, abs=1e-10)
+
+def test_total_cable():
+    export = ElectricalDesign(config)
+    export.run()
+
+    length = 0.02 + 3 + 30
+    length += length * 0.01
+    mass = length * export.cable.linear_density
+    assert export.total_mass == pytest.approx(mass *11, abs=1e-10)
+    assert export.total_length == pytest.approx(length * 9, abs=1e-10)
+
+def test_cables_property():
+    export = ElectricalDesign(config)
+    export.run()
+
+    assert (
+        export.sections_cables == export.cable.name
+    ).sum() == export.num_cables
+
+def test_cable_lengths_property():
+    export = ElectricalDesign(config)
+    export.run()
+
+    cable_name = export.cable.name
+    assert (
+        export.cable_lengths_by_type[cable_name] == export.length
+    ).sum() == export.num_cables
+
+def test_total_cable_len_property():
+    export = ElectricalDesign(config)
+    export.run()
+
+    cable_name = export.cable.name
+    assert export.total_cable_length_by_type[cable_name] == pytest.approx(
+        export.total_length, abs=1e-10
+    )
+
+def test_design_result():
+    export = ElectricalDesign(config)
+    export.run()
+
+    _ = export.cable.name
+    cables = export.design_result["export_system"]["cable"]
+
+    assert cables["sections"] == [export.length]
+    assert cables["number"] == 9
+    assert cables["linear_density"] == export.cable.linear_density
+
+
+def test_floating_length_calculations():
+
+    base = deepcopy(config)
+    base["site"]["depth"] = 250
+    base["export_system_design"]["touchdown_distance"] = 0
+
+    sim = ElectricalDesign(base)
+    sim.run()
+
+    base_length = sim.total_length
+
+    with_cat = deepcopy(config)
+    with_cat["site"]["depth"] = 250
+
+    new = ElectricalDesign(with_cat)
+    new.run()
+
+    assert new.total_length < base_length
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
```

Comment:
Remove extra lines. I should get you set up with pre-commit hooks sometime. They help a lot with automating random formatting things like this.